dcos/http.py

```diff
@@ -13,7 +13,14 @@
 
 logger = util.get_logger(__name__)
 
-DEFAULT_TIMEOUT = 5
+DEFAULT_TIMEOUT = 15
```

Comment:
Let's make sure that the public documentation matches this: https://jira.mesosphere.com/browse/DCOS_OSS-1957.

Reply:
Why 15 seconds?

I built a binary from this branch and tested it with the issue I presented on https://jira.mesosphere.com/browse/DCOS_OSS-1988. The problem remained:

```
Adam@MacBook-Pro ~/D/m/m/d/cli> time dist/dcos --log-level=debug --debug package install --cli dcos-enterprise-cli --yes
MainThread: 2017-12-18 19:36:04,525 dcos/http.py:_request:155 - Sending HTTP ['get'] to ['https://172.17.0.2/capabilities']: {'Content-Type': 'application/vnd.dcos.capabilities+json;charset=utf-8;version=v1', 'Accept': 'application/vnd.dcos.capabilities+json;charset=utf-8;version=v1'}
MainThread: 2017-12-18 19:36:04,531 urllib3/connectionpool.py:_new_conn:824 - Starting new HTTPS connection (1): 172.17.0.2
send: b'GET /capabilities HTTP/1.1\r\nHost: 172.17.0.2\r\nConnection: keep-alive\r\nUser-Agent: python-requests/2.18.4\r\nAccept: application/vnd.dcos.capabilities+json;charset=utf-8;version=v1\r\nAccept-Encoding: gzip, deflate\r\nContent-Type: application/vnd.dcos.capabilities+json;charset=utf-8;version=v1\r\nAuthorization: token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJ1aWQiOiJhZG1pbiIsImV4cCI6MTUxNDA1NDc4MX0.IBOY8X8Bc7UmzalNMU51MowtM25JOhr8CPopcaAgqnAbKEJPslpMplPE_rZep39qGqj_ZueZCMKEkhTGAkgMsDpqIBgm1Wx-p6Cy_rg6BpVqVg3q-RBvnQyc1D4hUBN6BM5RLoPrWo1wwHGveuQ-RRE9hodXkr1f4U9kId2lQLtxYDN6DSeyLTI78aLsrFS7gGdNxSshTkAemAAdpsoQVgPg6MkM2aohcVu9XR9pGcGvpek266K7zDUMZWg0Gz0ocps_0c1JaQYhDaDIpfWaaM0lzQUoTEU3Q4Y04KyXJM4rr1qEhi9atFA26gfIRY2V00vYs-lWMRRxbCFEsqzRKQ\r\n\r\n'
reply: 'HTTP/1.1 200 OK\r\n'
MainThread: 2017-12-18 19:36:04,691 urllib3/connectionpool.py:_make_request:396 - https://172.17.0.2:443 "GET /capabilities HTTP/1.1" 200 144
MainThread: 2017-12-18 19:36:04,695 dcos/http.py:_request:185 - Received HTTP response [200]: {'Server': 'openresty/1.13.6.1', 'Date': 'Mon, 18 Dec 2017 19:36:34 GMT', 'Content-Length': '144', 'Content-Type': 'application/vnd.dcos.capabilities+json;charset=utf-8;version=v1', 'Connection': 'keep-alive'}
MainThread: 2017-12-18 19:36:04,695 dcos/util.py:timer:609 - duration: dcos.http._request: 0.17s
MainThread: 2017-12-18 19:36:04,702 dcos/http.py:_request:155 - Sending HTTP ['post'] to ['https://172.17.0.2/package/describe']: {'Content-Type': 'application/vnd.dcos.package.describe-request+json;charset=utf-8;version=v1', 'Accept': 'application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3'}
MainThread: 2017-12-18 19:36:04,705 urllib3/connectionpool.py:_new_conn:824 - Starting new HTTPS connection (1): 172.17.0.2
header: Server header: Date header: Content-Type header: Content-Length header: Connection send: b'POST /package/describe HTTP/1.1\r\nHost: 172.17.0.2\r\nConnection: keep-alive\r\nUser-Agent: python-requests/2.18.4\r\nAccept: application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3\r\nAccept-Encoding: gzip, deflate\r\nContent-Type: application/vnd.dcos.package.describe-request+json;charset=utf-8;version=v1\r\nContent-Length: 38\r\nAuthorization: token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJ1aWQiOiJhZG1pbiIsImV4cCI6MTUxNDA1NDc4MX0.IBOY8X8Bc7UmzalNMU51MowtM25JOhr8CPopcaAgqnAbKEJPslpMplPE_rZep39qGqj_ZueZCMKEkhTGAkgMsDpqIBgm1Wx-p6Cy_rg6BpVqVg3q-RBvnQyc1D4hUBN6BM5RLoPrWo1wwHGveuQ-RRE9hodXkr1f4U9kId2lQLtxYDN6DSeyLTI78aLsrFS7gGdNxSshTkAemAAdpsoQVgPg6MkM2aohcVu9XR9pGcGvpek266K7zDUMZWg0Gz0ocps_0c1JaQYhDaDIpfWaaM0lzQUoTEU3Q4Y04KyXJM4rr1qEhi9atFA26gfIRY2V00vYs-lWMRRxbCFEsqzRKQ\r\n\r\n'
send: b'{"packageName": "dcos-enterprise-cli"}'
reply: 'HTTP/1.1 200 OK\r\n'
MainThread: 2017-12-18 19:36:15,758 urllib3/connectionpool.py:_make_request:396 - https://172.17.0.2:443 "POST /package/describe HTTP/1.1" 200 1346
MainThread: 2017-12-18 19:36:15,762 dcos/http.py:_request:185 - Received HTTP response [200]: {'Server': 'openresty/1.13.6.1', 'Date': 'Mon, 18 Dec 2017 19:36:45 GMT', 'Content-Length': '1346', 'Content-Type': 'application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3', 'Connection': 'keep-alive'}
MainThread: 2017-12-18 19:36:15,762 dcos/util.py:timer:609 - duration: dcos.http._request: 11.06s
header: Server header: Date header: Content-Type header: Content-Length header: Connection By Deploying, you agree to the Terms and Conditions https://mesosphere.com/catalog-terms-conditions/#certified-services
Installing CLI subcommand for package [dcos-enterprise-cli] version [1.3.0]
MainThread: 2017-12-18 19:36:15,773 dcos/util.py:ensure_dir_exists:203 - Creating directory: '/Users/Adam/.dcos/clusters/edfc460f-77b4-476e-993a-c0910aa5f3e9/subcommands/dcos-enterprise-cli'
MainThread: 2017-12-18 19:36:15,800 dcos/http.py:_request:155 - Sending HTTP ['get'] to ['https://172.17.0.2/package/resource?url=https://downloads.mesosphere.io/cli/binaries/darwin/x86-64/1.3.0/da2e75e78079662dd9940c7e848770ad8e55c97124a57a2d5786a679e6dfe6d0/dcos-enterprise-cli']: {'Accept': 'application/json'}
MainThread: 2017-12-18 19:36:15,804 urllib3/connectionpool.py:_new_conn:824 - Starting new HTTPS connection (1): 172.17.0.2
send: b'GET /package/resource?url=https://downloads.mesosphere.io/cli/binaries/darwin/x86-64/1.3.0/da2e75e78079662dd9940c7e848770ad8e55c97124a57a2d5786a679e6dfe6d0/dcos-enterprise-cli HTTP/1.1\r\nHost: 172.17.0.2\r\nConnection: keep-alive\r\nUser-Agent: python-requests/2.18.4\r\nAccept: application/json\r\nAccept-Encoding: gzip, deflate\r\nAuthorization: token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJ1aWQiOiJhZG1pbiIsImV4cCI6MTUxNDA1NDc4MX0.IBOY8X8Bc7UmzalNMU51MowtM25JOhr8CPopcaAgqnAbKEJPslpMplPE_rZep39qGqj_ZueZCMKEkhTGAkgMsDpqIBgm1Wx-p6Cy_rg6BpVqVg3q-RBvnQyc1D4hUBN6BM5RLoPrWo1wwHGveuQ-RRE9hodXkr1f4U9kId2lQLtxYDN6DSeyLTI78aLsrFS7gGdNxSshTkAemAAdpsoQVgPg6MkM2aohcVu9XR9pGcGvpek266K7zDUMZWg0Gz0ocps_0c1JaQYhDaDIpfWaaM0lzQUoTEU3Q4Y04KyXJM4rr1qEhi9atFA26gfIRY2V00vYs-lWMRRxbCFEsqzRKQ\r\n\r\n'
MainThread: 2017-12-18 19:36:30,916 dcos/http.py:_request:177 - HTTP Timeout
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 387, in _make_request
  File "<string>", line 2, in raise_from
  File "urllib3/connectionpool.py", line 383, in _make_request
  File "http/client.py", line 1198, in getresponse
  File "http/client.py", line 297, in begin
  File "http/client.py", line 258, in _read_status
  File "socket.py", line 576, in readinto
  File "ssl.py", line 937, in recv_into
  File "ssl.py", line 799, in read
  File "ssl.py", line 583, in read
socket.timeout: The read operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 440, in send
  File "urllib3/connectionpool.py", line 639, in urlopen
  File "urllib3/util/retry.py", line 357, in increment
  File "urllib3/packages/six.py", line 686, in reraise
  File "urllib3/connectionpool.py", line 601, in urlopen
  File "urllib3/connectionpool.py", line 389, in _make_request
  File "urllib3/connectionpool.py", line 309, in _raise_timeout
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='172.17.0.2', port=443): Read timed out. (read timeout=15)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "dcos/http.py", line 164, in _request
  File "requests/api.py", line 58, in request
  File "requests/sessions.py", line 508, in request
  File "requests/sessions.py", line 618, in send
  File "requests/adapters.py", line 521, in send
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='172.17.0.2', port=443): Read timed out. (read timeout=15)
MainThread: 2017-12-18 19:36:30,918 dcos/util.py:timer:609 - duration: dcos.http._request: 15.12s
Request to URL [https://172.17.0.2/package/resource?url=https://downloads.mesosphere.io/cli/binaries/darwin/x86-64/1.3.0/da2e75e78079662dd9940c7e848770ad8e55c97124a57a2d5786a679e6dfe6d0/dcos-enterprise-cli] timed out.
       27.78 real         0.84 user         0.62 sys
```

That isn't to say that this change is not part of a meaningful change, but I'd like to know of the reason for changing from 5 to 15 - are there requests which take somewhere between those values?